### DISCUSSION TITLE
@click.outside & @keyup.escape.window bug

### DISCUSSION THREAD

**Discussion by ahmu83** [alpinejs/alpine#2385] [1. Help]
**@click.outside** & **@keyup.escape.window** are triggered 5 times. Not sure if the approach is right for adding the **@click.outside** & **@keyup.escape.window** directives. Here is an example:

```
<div x-data="clickOutsideTest">
  <table>
    <thead>
      <tr>
        <th>#</th>
        <th>Log Data</th>
      </tr>
    </thead>
    <tbody>
      <template x-for="record, index in records" :key="record.id">
        <tr :key="record.id">
          <td x-text="record.id"></td>
          <td>

            <textarea 
              x-data="{focus: false}" 
              @dblclick="addFocus" 
              :value="record.log_data" 
              :class="focus ? 'focus' : ''"
              @click.outside="removeFocus"
              @keyup.escape.window="removeFocus"></textarea>

          </td>
        </tr>
      </template>
    </tbody>
  </table>

  <script>
  function clickOutsideTest() {
    return {
      records: [
        {id: 1, log_data: 'asdfasdf'},
        {id: 2, log_data: 'asdfasdf'},
        {id: 3, log_data: 'asdfasdf'},
        {id: 4, log_data: 'asdfasdf'},
        {id: 5, log_data: 'asdfasdf'},
      ],
      addFocus(e) {

        console.log('addFocus');

        this.focus = true;

      },
      removeFocus() {

        console.log('removeFocus');

        this.focus = false;

      }

    }
  }
  </script>

  <style>
  textarea.focus {
    outline: 1px solid red;
    outline-offset: 2px;
  }
  </style>

</div>
```

**Comment by SimoTod** [comment]
It should be logged as help not bug...
However, you have 5 textareas each one with a click outside so, when you click in a different point of the page, each one will run the function. 

If you want to avoid that, you need to check that the event is originated by the current textarea.

Your code won't work anyway, you are trying to update focus on the parent component where it's not defined (it's defined in each children but you are not doing it right).

Finally, don't do it. There's a css selector that does what you need without having to write custom javascript code.

**Reply by ahmu83** [reply]
Thank you for the answer

**Reply by ahmu83** [reply]
Can you elaborate on using the click outside properly for scenarios like this?
